packages/task/src/browser/task-configurations.ts

```diff
@@ -0,0 +1,257 @@
+    protected readonly TASKFILEPATH = '.theia';
+    /** task configuration file name */
+    protected readonly TASKFILE = 'tasks.json';
+
+    protected client: TaskConfigurationClient | undefined = undefined;
+
+    constructor(
+        @inject(ILogger) protected readonly logger: ILogger,
+        @inject(FileSystemWatcherServer) protected readonly watcherServer: FileSystemWatcherServer,
+        @inject(FileSystem) protected readonly fileSystem: FileSystem
+    ) {
+        this.toDispose.push(watcherServer);
+        watcherServer.setClient({
+            onDidFilesChanged: changes =>
+                this.onDidTaskFileChange(changes)
+                    .then(() => {
+                        if (this.client !== undefined) {
+                            this.client.taskConfigurationChanged(this.getTaskLabels());
+                        }
+                    })
+                    .catch(err => {
+                        this.logger.error(err);
+                    })
+        });
+
+        this.toDispose.push(Disposable.create(() => {
+            this.tasksMap.clear();
+            this.client = undefined;
+        }));
+    }
+
+    setClient(client: TaskConfigurationClient) {
+        this.client = client;
+    }
+
+    dispose() {
+        this.toDispose.dispose();
+    }
+
+    watchWorkspaceConfiguration(wsRootUri: string): Promise<void> {
+        return this.watchConfigurationFile(wsRootUri);
+    }
+
+    protected watchConfigurationFile(wsRootUri: string): Promise<void> {
+        return new Promise(resolve => {
+            let configFile: string;
+            this.getConfigFile(wsRootUri)
+                .then(uri => {
+                    // start watching that config file
+                    configFile = uri;
+                    if (this.watchedConfigFileUris.indexOf(configFile) === -1) {
+                        this.watchedConfigFileUris.push(configFile);
+                        this.watchConfigFile(configFile);
+                        return;
+                    }
+
+                })
+                .then(() => {
+                    this.readConfig(configFile)
+                        .then(tasks => {
+                            for (const task of tasks) {
+                                this.tasksMap.set(task.label, task);
+                            }
+                            resolve();
+                        })
+                        .catch(err => {
+                            this.logger.error(`Workspace: ${err}`);
+                        });
+                })
+                .catch(err => {
+                    this.logger.error(`Workspace: ${err}`);
+                });
+        });
+    }
+
+    protected readConfig(uri: string): Promise<TaskOptions[]> {
+        return new Promise((resolve, reject) => {
+            const tasksFound: TaskOptions[] = [];
+
+            // read tasks defined in tasks.json
+            this.readTasks(uri)
```

Comment:
just return this expression, or use async/await

Reply:
I returned the expression.